### PR TITLE
tests: remove ubuntu lunar support

### PR DESCRIPTION
**Pull request by sergiocazzolato**
Ubuntu lunar is EoL

### CODE REVIEW

File: tests/lib/external/snapd-testing-tools/tests/os.query/task.yaml [hunk 1]
```diff
@@ -163,7 +163,6 @@ execute: |
         opensuse-15.5-64)
             os.query is-opensuse
             os.query is-opensuse 15.5
-            ! os.query is-opensuse tumbleweed
```
Comment (olivercalder) [inline tests/lib/external/snapd-testing-tools/tests/os.query/task.yaml:166]: Is this a drive-by fix? Seems unrelated to lunar

File: tests/main/desktop-portal-screenshot/task.yaml [hunk 2]
```diff
@@ -57,7 +57,7 @@ execute: |
     # Taking a screenshot is not supported in kinetic and lunar yet.
     # The error displayed is the following:
     # dbus.exceptions.DBusException: org.freedesktop.DBus.Error.UnknownMethod: No such interface \u201corg.freedesktop.portal.Screenshot\u201d on object at path /org/freedesktop/portal/desktop
-    if os.query is-ubuntu-ge 23.04; then
+    if os.query is-ubuntu-ge 23.10; then
```
Comment (olivercalder) [inline tests/main/desktop-portal-screenshot/task.yaml:60]: "kinetic and lunar yet" probably should be changed to just "mantic" now.

File: tests/main/uc20-create-partitions-encrypt/task.yaml [hunk 3]
```diff
@@ -53,7 +53,7 @@ prepare: |
     echo "Get the UC20 gadget"
     if os.query is-focal; then
         snap download --channel=20/edge pc
-    elif os.query is-ubuntu 22.04 || os.query is-ubuntu 23.04; then
+    elif os.query is-ubuntu 22.04; then
```
Comment (olivercalder) [inline tests/main/uc20-create-partitions-encrypt/task.yaml:56]: No need for mantic or noble here? Was this mistakenly tested on lunar before? Not sure why a core test would have run on lunar.

**Review by olivercalder — COMMENTED** [review]
Looks good, just a few small questions.

**Review by MiguelPires — APPROVED** [review]
lgtm

File: tests/lib/external/snapd-testing-tools/tests/os.query/task.yaml (hunk 1 above)
Comment (sergiocazzolato) [reply tests/lib/external/snapd-testing-tools/tests/os.query/task.yaml:166]: yes, but I merged the changes in snapd-testing-tools project and it was already merged in main branch

[the remark this answers is not in the document]
Comment (sergiocazzolato) [reply tests/lib/external/snapd-testing-tools/tests/os.query/task.yaml:166]: yes, agree, but it came with the changes when I pulled from snapd-testing-tools

File: tests/main/desktop-portal-screenshot/task.yaml (hunk 2 above)
Comment (sergiocazzolato) [reply tests/main/desktop-portal-screenshot/task.yaml]: updated. Thanks for the review

**Review by olivercalder — APPROVED** [review]
LGTM, thanks.

I see there are static checks failures (missing `details` fields ?), is this expected/unrelated?

File: tests/main/uc20-create-partitions-encrypt/task.yaml (hunk 3 above)
Comment (alfonsosanchezbeato) [reply tests/main/uc20-create-partitions-encrypt/task.yaml:64]: This looks like a UC20+ test. Maybe it would make sense for hybrid systems as well, but probably that would require some changes to the test, so for the moment I would not run it for non-UC.

Comment (sergiocazzolato) [reply tests/main/uc20-create-partitions-encrypt/task.yaml:64]: ok, thanks for the clarification, I'll create a pr to support uc24 when we start running tests there
